docs/Internals.md

```diff
@@ -254,6 +254,14 @@ withSecretBytes(function<byte[], type> functionWithSecret) {
 }
 ```
 
+##### Concurrent Access
+
+The `withSecretBytes` pseudocode above is not thread-safe code as written. A thread could disable the memory access as
+another thread attempts to read the secret, which would result in a SIGSEGV signal to the process. Some form of thread
+safety is needed to guard against this. In the Java reference implementation (as of this writing) this is implemented
```

Comment:
do we need to list `as of this writing`? 
Would make the reader think if the documentation is actually up to date or not.

Reply:
the intent of that bit was so this doesn't end up being inaccurate in case the implementation did change and we forgot to update this. it was meant to just serve as an example of how it could be implemented. can remove if there is still strong objection

Reply:
i suggest we keep it

Reply:
I am still doubtful about it. The documentation should always be accurate regardless of any code change that we make.
If we are documenting something, it means it is there. 

Once we update the code, we can create a ticket/issue so that we don't forget updating. IMO `as of this writing` is not providing the reader any information.

Reply:
again, this wasn't intended to document the code, so think we're splitting hairs here. i'm going to reword this to something more generic to make everyone happy